File: 0078-subsets/0078-subsets.java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> res = new ArrayList<>();
        int n = nums.length;
        for (int i = 0; i < (1 << n); i++) {
            List<Integer> temp = new ArrayList<>();
            for (int j = 0; j < n; j++) {
                if ((i & (1 << j)) != 0) {
                    temp.add(nums[j]);
                }
            }
            res.add(temp);
        }
        return res;
    }
}

File: 0078-subsets/0078-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = []
        def gen(idx, curr):
            if idx == len(nums):
                res.append(curr[:])
                return 
            
            gen(idx + 1, curr)
            
            curr.append(nums[idx])
            gen(idx + 1, curr)
            curr.pop()
            
        
        gen(0, [])
        
        return res
            
        
            
            